### PR TITLE
#1 apply figma design to header

### PR DESCRIPTION
**Pull request by AyanoNK**
https://github.com/CamelliaCommunity/wiki-frontend/issues/1

This PR is to apply [Figma's header design](https://www.figma.com/file/MAtikPPpURkXzY9rd3glQk/Camellia-Wiki?type=design&node-id=1440-913&mode=design&t=D0YdakSPCkc8uDhN-0) to the web layout.

...Hopefully, nothing breaks.

https://github.com/CamelliaCommunity/wiki-frontend/assets/22826510/45b5502f-5680-4e08-9565-f732d88f34a2

### CODE REVIEW

**Comment by flustix** [comment]
First of all, indentation is wrong. Please use 4 spaces for indentation.

If you are using vscode you can change this in the bottom right:
image

File: src/components/nav/NavBar.vue [hunk 1]
```diff
@@ -1,118 +1,115 @@
...
-					</NavBarButton>
-					<!--if there ever gets anything added here, give it the 'middle' attribute-->
-					<NavBarButton url="https://discord.gg/camellia" last>
-						<PhDiscordLogo :size="28" />
-					</NavBarButton>
-				</div>
-			</div>
-		</div>
+  <nav class="w-full fixed z-20 bg-background-3 py-2">
+    <div class="navbar flex justify-between max-w-screen-lg mx-auto">
```
Comment (flustix) [inline src/components/nav/NavBar.vue:49]: This navbar class isn't used in the css.

File: src/components/nav/NavBar.vue [hunk 2]
```diff
@@ -1,118 +1,115 @@
...
+        </div>
+      </div>
+    </div>
+  </nav>
+  <Transition name="slidedown-fade">
+    <nav
+      v-if="isOpen"
+      class="content fixed top-16 bg-background-1 mx-4 w-full rounded-t-2xl"
+    >
+      <div class="navbar flex justify-between max-w-screen-lg mx-auto py-2">
```
Comment (flustix) [inline src/components/nav/NavBar.vue:75]: Same thing here.

File: src/components/nav/NavBar.vue [hunk 3]
```diff
@@ -1,118 +1,115 @@
...
+          <RouterLink
+            class="hover:text-accent text-lg font-light"
+            v-for="link in rightLinkList"
+            :to="link.route"
+            exact
+            >{{ link.name }}</RouterLink
+          >
+        </div>
+      </div>
+      <!-- This w-[calc(100vw-2.125rem)] is too hacky. I think it is because of the page layout configuration. -->
```
Comment (flustix) [inline src/components/nav/NavBar.vue:96]: Maybe turn this into a variable? Because the same value is used by nav.content

**Review by flustix — CHANGES_REQUESTED** [review]

File: src/components/nav/NavBar.vue (hunk 1 above)
Comment (AyanoNK) [reply src/components/nav/NavBar.vue]: Oh, you're right. Forgot about it.

File: src/components/nav/NavBar.vue (hunk 2 above)
Comment (AyanoNK) [reply src/components/nav/NavBar.vue]: Oh, you're right. Forgot about it.

**Comment by AyanoNK** [comment]
> First of all, indentation is wrong. Please use 4 spaces for indentation.
> 
> If you are using vscode you can change this in the bottom right:

It was not VSCode itself, but Prettier. I added a conf file to specify it to four so that other people who go in that have Prettier don't have the same problem.

File: src/components/nav/NavBar.vue (hunk 3 above)
Comment (AyanoNK) [reply src/components/nav/NavBar.vue]: Sounds good.

**Comment by flustix** [comment]
Looks fine enough for me now. Gonna have @Aisuruneko take a look at it then then maybe merge

**Review by Aisuruneko — APPROVED** [review]
is good now

**Pull request merged**